nemo/collections/asr/metrics/wer.py

```diff
@@ -1236,20 +290,18 @@ def update(
         scores = 0
         references = []
         with torch.no_grad():
-            # prediction_cpu_tensor = tensors[0].long().cpu()
+            tgt_lenths_cpu_tensor = targets_lengths.long().cpu()
             targets_cpu_tensor = targets.long().cpu()
-            tgt_lenths_cpu_tensor = target_lengths.long().cpu()
-
+            # check batch_dim_index is first dim
+            if self.batch_dim_index != 0:
+                targets_cpu_tensor = move_dimension_to_the_front(targets_cpu_tensor, self.batch_dim_index)
             # iterate over batch
             for ind in range(targets_cpu_tensor.shape[0]):
                 tgt_len = tgt_lenths_cpu_tensor[ind].item()
                 target = targets_cpu_tensor[ind][:tgt_len].numpy().tolist()
                 reference = self.decoding.decode_tokens_to_str(target)
                 references.append(reference)
-
-            hypotheses, _ = self.decoding.ctc_decoder_predictions_tensor(
-                predictions, predictions_lengths, fold_consecutive=self.fold_consecutive
-            )
+            hypotheses, _ = self.decode(predictions, predictions_lengths)
```

Comment:
Need to pass fold consecutive here

Reply:
We never call it in CTC models. So it just fall on default. Let me check the ssl models.